samples/graphene-hello/preload/README.md

```diff
@@ -19,31 +19,24 @@ Then get `mr_enclave` from the build output and set it as `UniqueID` in `manifes
 Note that compared to the `spawn` method for this sample, the argv arguments are defined during the build and cannot be changed afterwards without rebuilding your Graphene application. To adjust the provisioned argv arguments, you can change the entry `hello.argv` in the Makefile.
 
 ## Run
-We assume that the Coordinator is run with the following environment variables:
-
-- EDG_COORDINATOR_MESH_ADDR=localhost:2001
-- EDG_COORDINATOR_CLIENT_ADDR=localhost:4433
-- EDG_COORDINATOR_DNS_NAMES=localhost
-- EDG_COORDINATOR_SEAL_DIR=$PWD
+Next, use the `erthost` command to start the Coordinator in a local simulated enclave:
+```sh
+OE_SIMULATION=1 erthost ../../../build/coordinator-enclave.signed
```

Comment:
Why OE_SIMULATION=1? 

Graphene specifically requires SGX-enabled hardware. You cannot run Graphene applications in simulation mode. And if you already run the application on an SGX-enabled system (which I think we should assume for the samples), why specifically run the Cooridinator in simulation mode?

Reply:
> Graphene specifically requires SGX-enabled hardware. You cannot run Graphene applications in simulation mode. And if you already run the application on an SGX-enabled system, why specifically run the Coordinator in simulation mode?

You can run Graphene without SGX, but we're currently not doing this in our Makefile.
Most people probably won't have SGX enabled on their machines (and if they don't run in Azure, they won't have a DCAP infrastructure to verify quotes), hence, the incentive for OE_SIMULATION makes sense to me. We can argue that people TRYING our examples are more likely to have access to some SGX machine.

> however for the samples I think it is important to demonstrate that with the UniqueID (or other parameters) pinned in the manifest, you cannot simply modify your application as you please. This only works properly in non-simulation mode.

I see your point. For me, it's a compromise between keeping the bar to interact with our samples as low as possible and actually having meaningful samples.
I agree with @Nirusu that the current setup of Coordinator in simulation mode and Graphene in SGX mode does not make sense because we get neither of those traits.
I'd say we remove the `OE_SIMULATION` because building the Graphene enclave requires a working Graphene environment which most people will have as their "Cloud Development Setup". It might be useful for us to have something similar in our docs: https://graphene.readthedocs.io/en/latest/cloud-deployment.html

Reply:
Maybe we should provide an overview of the examples sorted by their requirements with categories like "no-sgx", "sgx-vm", "sgx-cluster"?
I like the idea of having some "prerequisite setup" tutorials as @m1ghtym0 suggested.

Reply:
> You can run Graphene without SGX, but we're currently not doing this in our Makefile.

Ah, now this was a bad oversimplification on my part. Yes, you can actually run Graphene without SGX, but you need to have build the entire tool set once without SGX=1 initially (otherwise you're missing the non-SGX pal-loader by default) and we would need to adjust our Makefile.

What I was meant to say here is that you cannot simply interchange it _that_ easily as with our tools by just running something prebuilt in simulation mode. It has to be already considered for when setting up / building Graphene, thus you would go down an entirely different path (non-SGX build) during the initial approach anyway.

> For me, it's a compromise between keeping the bar to interact with our samples as low as possible and actually having meaningful samples.

I agree here. But I think it is important that we do not simply append OE_SIMULATION=1 as default, as we basically bypass security here. We basically do not even run correctly on SGX, we just simulate it. Now I get that the DCAP infrastructure is still immature, users might likely not have the required environment here even if their hardware supports SGX etc., but I think we should aim for two things:

- Whenever we can fix certain situations in our tooling without a security risk, we should do it in tooling. 
- Above that, we should educate the user here about any insecurities for going down "the easy way"

We advertise Marblerun as a SGX-cluster solution. I do not think it would make sense to assume by default that we do not care about SGX anymore just to make them errors go away. I agree that for initial setup in a testing environment, it properly does not matter to do everything 100% correctly. But then the user should know about it, or have to follow a (slightly) different tutorial path. But we should not go for "insecure by default, afterwards switch to secure whenever you're done".

> Maybe we should provide an overview of the examples sorted by their requirements with categories like "no-sgx", "sgx-vm", "sgx-cluster"?

I think @katexochen 's suggestion to have different categories here might be good, though I hope we can keep this short but sweet. Otherwise, I feat that it could result in TL;DR if it gets to long. 

But I think we need to educate the user somehow on the current landscape of SGX, DCAP, what is/should be correct and what not, but allows them testing at least 90% of the feature set and UX. After reading tutorials & running the samples, they should be at least somewhat confident in using our tools.

I think if you go for Marblerun, you properly care about confidential computing in someway. Now, if you bypass everything, you should at least know that this is not quite intended to work this way it is supposed too, but that we just choose it for initial simplicity. 

Therefore, please do not set this as default, but rather as an alternative way :)

Reply:
> Therefore, please do not set this as default, but rather as an alternative way :)

just to clarify, this is nothing I introduced, the sample previous used the simulation mode, since it referred to the [BUILD.md](https://github.com/edgelesssys/marblerun/blob/master/BUILD.md#run-the-coordinator). I just copied the command from there.

In my opinion, tutorials should be as linear as possible, therefore I wouldn't add alternative ways. They should describe a single exemplary scenario.

I've deleted the OE_SIMULATION=1 in the Graphene examples and added a line mentioning that SGX is needed. I've also added an explanation and security note to the examples that don't use SGX (the C++ and Hello World example).

Reply:
> just to clarify, this is nothing I introduced, the sample previous used the simulation mode, since it referred to the BUILD.md. I just copied the command from there.

Yup, I get that. But I think the BUILD.md [might need a note as we do for our quickstart guide](https://www.marblerun.sh/docs/getting-started/quickstart/), instead of just starting without any mention.

Though then it is a BUILD guide, and many of our tutorials assume starting with the prebuilt binaries. 

I think there's still plenty of room for improvement in our documentation. Or for the SGX-world in general, it is still confusing in most tools. And I think we can try to do better.

> In my opinion, tutorials should be as linear as possible, therefore I wouldn't add alternative ways.

I do not think it's too bad if done well. Especially for the libOS samples (Graphene, Occlum), their guides do something similar. Graphene has seperate non-SGX, SGX steps. [Occlum mentions shortly you can run this in simulation mode](https://github.com/occlum/occlum#hello-occlum) for non-SGX platforms, which is similar on how we do it on our website.

The note you added seems okay for me for this PR, I would be okay with it! 

For the future, I think it would be great if we could:

- Develop a small program which determines the current system's SGX capabilities (Does CPU support it? Is it set up correctly? Does DCAP work?)
- Give users a simple answer, give them a path for a "Getting Started" guide and our samples
- Expect requirements / limitations of possible paths
- Keep things as streamlined as possible, but make seperate trails for cases where we really cannot streamline it in a way it makes sense

And I think adding OE_SIMULATION=1 in a non-ideal scenario does not require too much alternative ways.

But yeah, for the sake of the current PR, I think such a small note for the "simple" non-libOS samples is okay, and for the libOS samples we could assume they require SGX hardware or we can assume that the user should know that many tools allow switching you between non-SGX/simulation mode and SGX-mode, because at this point they properly already went through their documentation to build their application.